components/gmail/sources/new-email-received-instant/new-email-received-instant.mjs

```diff
@@ -0,0 +1,389 @@
+import { PubSub } from "@google-cloud/pubsub";
+import { v4 as uuidv4 } from "uuid";
+import common from "../common/verify-client-id.mjs";
+
+export default {
+  key: "gmail-new-email-received-instant",
+  name: "New Email Received (Instant)",
+  description: "Emit new event when a new email is received.",
+  version: "0.0.1",
+  type: "source",
+  props: {
+    ...common.props,
+    http: "$.interface.http",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 24 * 60 * 60,
+      },
+      hidden: true,
+    },
+    db: "$.service.db",
+    topicType: {
+      type: "string",
+      label: "Type",
+      description: "Do you have an existing Pub/Sub topic, or would you like to create a new one?",
+      options: [
+        "existing",
+        "new",
+      ],
+      reloadProps: true,
+    },
+    label: {
+      propDefinition: [
+        common.props.gmail,
+        "label",
+      ],
+      default: "INBOX",
+      optional: true,
+    },
+    keyAlert: {
+      type: "alert",
+      alertType: "error",
+      content: `No Service Account Key JSON found. Please reconnect your Gmail account in Pipedream, and add a Service Account Key JSON.
+      \nIn order to receive real-time push notifications for changes to your email inbox, you will need to first enable the Google Cloud Pub/Sub API within your Google Cloud console.
+      \n
+      \n1. Navigate to the [Cloud Pub/Sub API](https://console.cloud.google.com/apis/library/pubsub.googleapis.com) and click “Enable.”
+      \n2. Under “API and Services”, click “Credentials”
+      \n3. Click “Create Credentials”, then “Service Account”
+      \n4. Name your service account, e.g. “Gmail PubSub Handler”, and optionally provide a description, e.g. “Used for configuring Pub/Sub API with Gmail on Pipedream.”
+      \n5. Click on your newly created service account. Under the heading “Keys”, click “Add Key” then “Create Key”, Key Type: JSON. The private key will be saved to your computer; be sure to store this securely. 
+      `,
+      hidden: true,
+    },
+    permissionAlert: {
+      type: "alert",
+      alertType: "error",
+      content: `Unable to grant publish permission to Gmail API service account.
+      \n1. Navigate to your [Google Cloud PubSub Topics List](https://console.cloud.google.com/cloudpubsub)
+      \n2. Select "View Permissions" for the topic you intend to use for this source.
+      \n3. Click "ADD PRINCIPAL"
+      \n4. Select "Pub/Sub Publisher" for the Role.
+      \n5. Enter \`serviceAccount:gmail-api-push@system.gserviceaccount.com\` as the principal. 
+      \n6. Click "Save"
+      `,
+      hidden: true,
+    },
+  },
+  async additionalProps(props) {
+    const isValidClientId = await this.checkClientId();
+    props.appAlert.alertType = isValidClientId
+      ? "info"
+      : "error";
+    props.appAlert.content = isValidClientId
+      ? "OAuth client ID is valid. You can use the component."
```

Comment:
I think we can omit the success case, and only throw a warning for the error case.